### PR TITLE
added logs

### PR DESCRIPTION
**Pull request by anshul-goyal-t**

### CODE REVIEW

File: fast_api_als/database/db_helper.py [hunk 1]
```diff
@@ -15,7 +15,12 @@
     
     write a commong function that logs this response code with appropriate context data
 """
+logging.basicConfig(filename='test.log', level=logging.INFO,
+format='%(asctime)s:%(levelname)s:%(message)s')
 
+def get_response_logger (res,parent_func):
```
Comment (nitish-ti) [inline fast_api_als/database/db_helper.py:21]: Implemented it right, but you're not using it.

File: fast_api_als/routers/lead_conversion.py [hunk 2]
```diff
@@ -44,17 +46,24 @@ def get_quicksight_data(lead_uuid, item):
...
+    try :
+        provider, role = get_user_role(token)
+    except:
+        logging.error("provider / role is undefined in post request of reset_authkey")
+    logging.info("provider and role in post request of authkey is "+provider+role)
     if 'lead_uuid' not in body or 'converted' not in body:
-        # throw proper HTTPException
+        raise HTTPException(
+            status_code=400,detail="leaduuid or converted is missing")
         pass
```
Comment (nitish-ti) [inline fast_api_als/routers/lead_conversion.py]: remove pass

File: fast_api_als/routers/submit_lead.py [hunk 3]
```diff
@@ -120,7 +127,8 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
             "message": "OEM data not found"
         }
     oem_threshold = float(fetched_oem_data['threshold'])
-
+    if not oem_threshold:
+        logging.warning("oem_threshold in post request of submit is undefined")
```
Comment (nitish-ti) [inline fast_api_als/routers/submit_lead.py]: You should not write "post request of submit" in each log statement.
It's vague, repetitive and have more length than your important message.

File: fast_api_als/routers/submit_lead.py [hunk 4]
```diff
@@ -132,13 +140,19 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
...
     # convert the enriched lead to ML input format
     ml_input = conversion_to_ml_input(model_input, make, dealer_available)
 
     # score the lead
     result = score_ml_input(ml_input, make, dealer_available)
-
+    if not result:
+        logging.warning("result in post request of submit is undefined")
+    else :
+        logging.info("result is good")
```
Comment (nitish-ti) [inline fast_api_als/routers/submit_lead.py]: if-else just for writing the log statement, not worth it.
shorten this.

File: fast_api_als/routers/submit_lead.py [hunk 5]
```diff
@@ -196,6 +210,10 @@ async def submit(file: Request, apikey: APIKey = Depends(get_api_key)):
                 'model': model
             }
         }
+        if not message:
+            logging.warning("message in post request of submit is undefined")
+        else :
+            logging.info("message is good in post request of submit")
```
Comment (nitish-ti) [inline fast_api_als/routers/submit_lead.py]: shorten this.

File: fast_api_als/routers/three_pl.py [hunk 6]
```diff
@@ -10,16 +10,29 @@
...
+    try :
+        provider, role = get_user_role(token)
+    except:
+        logging.error("provider / role is undefined in post request of reset_authkey")
+    logging.info("provider and role in post request of authkey is "+provider+role)
+    
     if role != "ADMIN" and (role != "3PL"):
-        pass
+        raise HTTPException(
+            status_code=401)
```
Comment (nitish-ti) [inline fast_api_als/routers/three_pl.py]: add proper message also

File: fast_api_als/utils/adf.py [hunk 7]
```diff
@@ -4,15 +4,22 @@
...
 
 # ISO8601 datetime regex
 regex = r'^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(\.[0-9]+)?(Z|[+-](?:2[0-3]|[01][0-9]):[0-5][0-9])?$'
 match_iso8601 = re.compile(regex).match
 zipcode_search = SearchEngine()
-
+if not match_iso8601:
+            logging.warning("match_iso8601 is not defined")
+else :
+            logging.info("match_iso8601 is defined")
```
Comment (nitish-ti) [inline fast_api_als/utils/adf.py]: ?

File: fast_api_als/utils/adf.py [hunk 8]
```diff
@@ -38,8 +45,15 @@ def is_nan(x):
...
-    # use exception handling
+    if not adf_xml:
+            logging.warning("adf_xml is not defined")
+    else :
+                logging.info("adf_xml is defined")
     obj = xmltodict.parse(adf_xml)
+    if not obj:
+            logging.warning("obj is not defined")
+    else :
+                logging.info("obj is defined")
```
Comment (nitish-ti) [inline fast_api_als/utils/adf.py]: what if xml was badly format and xmltodict.parse() throw exception?

Comment (nitish-ti) [reply fast_api_als/utils/adf.py]: handle it accordingly

File: fast_api_als/database/db_helper.py [hunk 9]
```diff
@@ -39,7 +44,7 @@ def insert_lead(self, lead_hash: str, lead_provider: str, response: str):
             'ttl': datetime.fromtimestamp(int(time.time())) + timedelta(days=constants.LEAD_ITEM_TTL)
         }
         res = self.table.put_item(Item=item)
-
+        get_response_logger(res,insert_lead)
```
Comment (nitish-ti) [inline fast_api_als/database/db_helper.py]: Shouldn't you be sending "insert_lead" ??

Comment (anshul-goyal-t) [reply fast_api_als/database/db_helper.py]: Right , my bad

**Pull request closed without merging**